pkg/build/api/types.go

```diff
@@ -25,6 +25,8 @@ const (
 	BuildJenkinsLogURLAnnotation = "openshift.io/jenkins-log-url"
 	// BuildJenkinsBuildURIAnnotation is an annotation holding a link to the Jenkins build
 	BuildJenkinsBuildURIAnnotation = "openshift.io/jenkins-build-uri"
+	// BuildMatchSourceURIAnnotation is an annotation on a Secret indicating the URIs for which it can be used
+	BuildMatchSourceURIAnnotation = "build.openshift.io/match-source-uri"
```

Comment:
migrating our code to use from one to the other(and in the process respecting both, not necessarily migrating the existing instances).  It's a migration, just not what you're thinking of as a data migration.

>  Do you see any of that as a reason to perpetuate a pattern which no longer serves us?

i don't accept your premise that the current pattern doesn't serve us.  it's work so far and i'm not aware of a single pain point it has introduced.

>  "forever be inconsistent with any new API group being developed"

i don't think it's reasonable to assume annotations are always going to fit nicely into a particular api group, so we're just creating a new arbitrary pattern that will break down.